compat-tests.el

```diff
@@ -1004,6 +1004,64 @@
       (should-equal compat-tests--global 1)
       (should-equal compat-tests--local 2)
       (should-not (boundp 'compat-tests--unexist)))))
+ 
+(ert-deftest compat-buffer-tests-inhibit-buffer-hooks ()
```

Comment:
I suggest to merge the two complex tests into one test function `compat-generate-new-buffer`. Then add a simpler separate test `compat-get-buffer-create`. I insist on having separate tests for each Compat-provided function.

Reply:
What did you have in mind for a test for `compat-get-buffer-create`? Would the test for `compat-get-buffer-create` be modeled off of the test for `compat-generate-new-buffer`?

Reply:
No, a simple test would be fine, given that we already test the function indirectly, i.e., just calling the function once with and once without the argument, checking that it returns a buffer with a new name. I want to make sure that the function is defined via Compat. That's all.